stactools_landsat/stactools/landsat/utils.py

```diff
@@ -0,0 +1,141 @@
+import datetime
+
+import dateutil
+import rasterio
+from pystac import Item, Link
+from rasterio import RasterioIOError
+from shapely.geometry import box, mapping, shape
+
+
+def _parse_date(in_date: str) -> datetime.datetime:
+    """
+    Try to parse a date and return it as a datetime object with no timezone
+    """
+    dt = dateutil.parser.parse(in_date)
+    return dt.replace(tzinfo=datetime.timezone.utc)
+
+
+def transform_mtl_to_stac(metadata: dict) -> Item:
+    """
+    Handle USGS MTL as a dict and return a STAC item.
+
+    NOT IMPLEMENTED
+
+    Issues include:
+        - There's no reference to UTM Zone or any other CRS info in the MTL
```

Comment:
Is the `PROJECTION_ATTRIBUTES` section not enough? It looks like maybe it doesn't say whether the utm zone is north or south... but I'd guess you could deduce it.

```bash
> aws s3 cp s3://usgs-landsat/collection02/level-2/standard/oli-tirs/2020/035/033/LC08_L2SP_035033_20200725_20200908_02_T2/LC08_L2SP_035033_20200725_20200908_02_T2_MTL.json - --request-payer | jq '.LANDSAT_METADATA_FILE.PROJECTION_ATTRIBUTES'
```

```json
{
  "MAP_PROJECTION": "UTM",
  "DATUM": "WGS84",
  "ELLIPSOID": "WGS84",
  "UTM_ZONE": "13",
  "GRID_CELL_SIZE_REFLECTIVE": "30.00",
  "GRID_CELL_SIZE_THERMAL": "30.00",
  "REFLECTIVE_LINES": "8001",
  "REFLECTIVE_SAMPLES": "7871",
  "THERMAL_LINES": "8001",
  "THERMAL_SAMPLES": "7871",
  "ORIENTATION": "NORTH_UP",
  "CORNER_UL_LAT_PRODUCT": "39.93756",
  "CORNER_UL_LON_PRODUCT": "-109.18018",
  "CORNER_UR_LAT_PRODUCT": "40.00431",
  "CORNER_UR_LON_PRODUCT": "-106.41877",
  "CORNER_LL_LAT_PRODUCT": "37.78038",
  "CORNER_LL_LON_PRODUCT": "-109.05554",
  "CORNER_LR_LAT_PRODUCT": "37.84221",
  "CORNER_LR_LON_PRODUCT": "-106.37634",
  "CORNER_UL_PROJECTION_X_PRODUCT": "142800.000",
  "CORNER_UL_PROJECTION_Y_PRODUCT": "4429200.000",
  "CORNER_UR_PROJECTION_X_PRODUCT": "378900.000",
  "CORNER_UR_PROJECTION_Y_PRODUCT": "4429200.000",
  "CORNER_LL_PROJECTION_X_PRODUCT": "142800.000",
  "CORNER_LL_PROJECTION_Y_PRODUCT": "4189200.000",
  "CORNER_LR_PROJECTION_X_PRODUCT": "378900.000",
  "CORNER_LR_PROJECTION_Y_PRODUCT": "4189200.000"
}
```

Reply:
Yes, looks like the corner projection coordinates and the grid cell size has everything to construct the affine. 
Will also need the EPSG code which can be derived (see https://github.com/sat-utils/sat-stac-landsat/blob/master/satstac/landsat/main.py#L196).

The only question I had is for other bands that are not at the nominal cell size. Thermal bands are distributed as upsampled versions to 30m, but the pan band (which is only available in the Level-1 products) is 15m. I think those projection coordinates are still valid, but may need to be shifted. Quick look shows the pan band as 2*x+1 rows/cols as opposed to the nominal 30m bands. So those projection coordinates may need to be shifted half a pixel, but I have not looked into this closely.

Reply:
It looked like the pan band is similarly offset by half a pixel (7.5m) compared to the MTL bounds in the local CRS.

Reply:
Ah, and the MTL I downloaded from here doesn't have `UTM_ZONE`.

https://landsatlook.usgs.gov/data/collection02/level-2/standard/oli-tirs/2020/081/119/LC08_L2SR_081119_20200101_20200823_02_T2/LC08_L2SR_081119_20200101_20200823_02_T2_MTL.json

I'll check another.

Reply:
S3 command to get it:
```
aws s3 cp --request-payer requester \
	s3://usgs-landsat/collection02/level-2/standard/oli-tirs/2020/081/119/LC08_L2SR_081119_20200101_20200823_02_T2/LC08_L2SR_081119_20200101_20200823_02_T2_MTL.json .
```

This file is `L2SR` whereas yours was `L2SP`. Surface reflectance vs. ???

I don't know why they're different...

Reply:
That MTL doesn't have a UTM zone because `MAP_PROJECTION` is `PS`, i.e. Polar Stereographic. I believe every MTL where the projection is `UTM` includes a `UTM_ZONE` key. I haven't worked with the PS projection before, so I'm not certain how to handle such MTL files. But I assume that there are only two relevant PS projections, one for the north pole and one for the south pole, and should be obvious which one from the latitude values.

```bash
> cat LC08_L2SR_081119_20200101_20200823_02_T2_MTL.json| jq '.LANDSAT_METADATA_FILE.PROJECTION_ATTRIBUTES'
```

```json
{
  "MAP_PROJECTION": "PS",
  "DATUM": "WGS84",
  "ELLIPSOID": "WGS84",
  "VERTICAL_LON_FROM_POLE": "0.00000",
  "TRUE_SCALE_LAT": "-71.00000",
  "FALSE_EASTING": "0",
  "FALSE_NORTHING": "0",
  "GRID_CELL_SIZE_REFLECTIVE": "30.00",
  "REFLECTIVE_LINES": "7671",
  "REFLECTIVE_SAMPLES": "7761",
  "ORIENTATION": "NORTH_UP",
  "CORNER_UL_LAT_PRODUCT": "-82.03706",
  "CORNER_UL_LON_PRODUCT": "98.46113",
  "CORNER_UR_LAT_PRODUCT": "-79.92551",
  "CORNER_UR_LON_PRODUCT": "96.67232",
  "CORNER_LL_LAT_PRODUCT": "-81.46766",
  "CORNER_LL_LON_PRODUCT": "112.64692",
  "CORNER_LR_LAT_PRODUCT": "-79.47133",
  "CORNER_LR_LON_PRODUCT": "108.16484",
  "CORNER_UL_PROJECTION_X_PRODUCT": "857100.000",
  "CORNER_UL_PROJECTION_Y_PRODUCT": "-127500.000",
  "CORNER_UR_PROJECTION_X_PRODUCT": "1089900.000",
  "CORNER_UR_PROJECTION_Y_PRODUCT": "-127500.000",
  "CORNER_LL_PROJECTION_X_PRODUCT": "857100.000",
  "CORNER_LL_PROJECTION_Y_PRODUCT": "-357600.000",
  "CORNER_LR_PROJECTION_X_PRODUCT": "1089900.000",
  "CORNER_LR_PROJECTION_Y_PRODUCT": "-357600.000"
}
```

Reply:
And now I feel a bit silly, @kylebarron! I haven't encountered Landsat scenes in polar stereographic before... I picked a bad example! hehe

Although it does demonstrate the extra case that needs to be handled in translating MTLs.

Reply:
I haven't worked with PS scenes either. I'm guessing you'd just need an if/else to switch between georeferencing for UTM and PS. But this should allow creating any image transform without reading the image I think (though I've already been wrong once 😄 )

Reply:
Yeah, we can add handling of the (presumably) two polar projections and many UTM zones separately... I'll find some example STAC files and add more tests this week.

Reply:
Followup to this conversation captured in https://github.com/stac-utils/stactools/issues/48 - feel free to correct or add to that issue!